warehouse/models/staging/audit/stg_audit__cloudaudit_googleapis_com_data_access.sql

```diff
@@ -0,0 +1,95 @@
+{{
+    config(
+        materialized='incremental',
+        incremental_strategy='insert_overwrite',
+        partition_by={
+            'field': 'date',
+            'data_type': 'date',
+            'granularity': 'day',
+        },
+        partitions=['current_date()'],
+        cluster_by='job_type',
+    )
+}}
+-- we should use https://cloud.google.com/bigquery/docs/reference/standard-sql/json_functions#parse_json when available
+
+-- Note these two source CTEs a direct reference instead of source, because a new table is created daily
+WITH latest AS (
+    {% set today = modules.datetime.date.today() %}
+
+    SELECT *
+    FROM cal-itp-data-infra.audit.cloudaudit_googleapis_com_data_access_{{ today.strftime('%Y%m%d') }}
+),
+
+everything AS (
+    {% set start_date = modules.datetime.date(year=2022, month=4, day=11) %}
+    {% set days = (modules.datetime.date.today() - start_date).days + 1 %}
+
+    {% for add in range(days) %}
+
+    {% set current = start_date + modules.datetime.timedelta(days=add) %}
+
+    SELECT *
+    FROM cal-itp-data-infra.audit.cloudaudit_googleapis_com_data_access_{{ current.strftime('%Y%m%d') }}
+    {% if not loop.last %}
+    UNION ALL
+    {% endif %}
+
+    {% endfor %}
+),
+
+data_to_process AS (
+    SELECT
+        *,
+        EXTRACT(DATE FROM timestamp) AS date,
+        -- do some initial renaming/unnesting here since we can't refer to same-SELECT aliases
+        protopayload_auditlog AS payload,
+        protopayload_auditlog.metadataJson AS metadata,
+
+        JSON_QUERY(protopayload_auditlog.metadataJson, '$.jobChange.job') AS job
+
+    {% if is_incremental() %}
+    FROM latest
```

Comment:
this is a new pattern for us (checking for incremental), not sure if it's worth flagging for a short talk through at some data services internal team meeting about cases where you'd do this / the rationale